### PR TITLE
Remove fuzz and return face id alongside element id (Replace stale #308)

### PR DESCRIPTION
**Pull request by Sichao25**
Resolved the merge conflict and updated the PR #308  to fit the new field API.

### CODE REVIEW

File: src/pcms/localization/point_search.h [hunk 1]
```diff
@@ -60,6 +124,19 @@ class PointLocalizationSearch
...
+   * This function provides a temporary solution to retrieve the original
+   * behavior of the point search, which previously returned the face ID
+   * regardless of which entity the search result belonged to. Many parts of
+   * the codebase still rely on this legacy behavior. After updating the point
+   * search to return the exact entity, this function can be called to retrieve
+   * the old behavior and ensure correctness. Long term, the implementation
+   * should be updated to properly handle the new result.
+   */
+  [[nodiscard]] virtual LO GetOwningElementId(const Result& result) const = 0;
+  [[nodiscard]] virtual Kokkos::View<LO*> GetOwningElementIds(
```
Comment (jacobmerson) [inline src/pcms/localization/point_search.h:111]: Can these GetOwningElementId functions be made into free functions?

Comment (Sichao25) [reply src/pcms/localization/point_search.h:111]: It is possible, but since we need to access some private members, more API changes such as getter functions are required.

Comment (jacobmerson) [reply src/pcms/localization/point_search.h:111]: Ok, so I think we should probably have a free function that given `mesh`, `dim`, `id` gives back the owning element. The member function can call this.

The free function version will be useful in a number of other places.

File: src/pcms/localization/point_search.h [hunk 2]
```diff
@@ -22,12 +60,76 @@ namespace detail
...
+ * (triangle) by computing the triangle area from two edge vectors using the
+ * parallelogram formula valid in any dimension:
+ *   area = 0.5 * sqrt(||a||^2 ||b||^2 - (a·b)^2)
+ *
+ * Template parameter N is the embedding dimension (e.g., 2 for planar, 3 for
+ * spatial), while the simplex is always a triangle (3 vertices).
+ */
+template <int N>
+[[nodiscard]] KOKKOS_FUNCTION inline Real
+distance_to_closest_edge_from_barycentric(const Omega_h::Matrix<N, 3>& coords,
```
Comment (HelenW42) [inline src/pcms/localization/point_search.h]: In the case when a point is on the infinite extension of one of the face's edges, this function will erroneously say the point is 0 units from an edge. Also, this method will not (as the comment states) extend to 3D: barycentric coordinates in 3D are the ratios of volumes of tetrahedra, thus the product of the ith coordinate and the height/altitude of the face is not in the distance of the point to the edge.

Comment (jacobmerson) [reply src/pcms/localization/point_search.h]: @HelenW42 did you end up having a fix for this function?

Comment (Sichao25) [reply src/pcms/localization/point_search.h]: I am working on a fix in 2D that computes the distance to the three edges and compares them, while keeping the original function as an efficient solution for points inside the element. I haven't implemented or thought much about the 3D case yet. Happy to hear any thoughts or if there are any known solutions.

Comment (HelenW42) [reply src/pcms/localization/point_search.h]: The attached PDF has my (rough) derivations of how to find whether a point is within a certain tolerance of an edge, vertex, and face in 2D and 3D from the barycentric coordinates alone. While I didn't include a specific fix for the collinear edge case, it's not that expensive an addition: if the point seems to have zero distance from the edge and is not within it, just check the distance from the vertices that define that edge (the derivation for which is in the file).
[bary_to_global_tolerance.pdf](https://github.com/user-attachments/files/29868937/bary_to_global_tolerance.pdf)

Comment (Sichao25) [reply src/pcms/localization/point_search.h]: Based on an offline discussion with Jacob, it should be okay to remove the function `distance_to_closest_edge_from_barycentric` from this PR since the current `GridPointSearch` class doesn't use it during the search process. I'll add a test case covering the edge case for the current approach. We can then address this barycentric-based approach in a separate issue or PR.

**Comment by Sichao25** [comment]
@jacobmerson  The point search function based on barycentric coordinates without reference has been removed.

**Review by jacobmerson — APPROVED** [review]
